apps/settings/js/modules/addon_manager.js

```diff
@@ -105,6 +108,108 @@ define(function(require) {
     });
   }
 
+  function renameAddon(addon, name) {
+    if (!name) {
+      return Promise.reject('no name given');
+    }
+
+    if (!isAddon(addon)) {
+      return Promise.reject('not an addon');
+    }
+
+    var manifest = getManifest(addon);
+    if (name === manifest.name) {
+      return Promise.reject('name is unchanged');
+    }
+
+    // To rename an addon we are creating a new DOMApplication that will have
+    // an updated name. Other manifest attributes are preserved.
+    return new Promise(function(resolve, reject) {
```

Comment:
The promise chain starts from `addon.export` so I think we can just return it without wrapping it with a new promise. Or is there any other reason of doing the wrapping?